main.go

```diff
@@ -45,6 +45,18 @@ func main() {
 			maxAge, _ := strconv.Atoi(maxAgeParams[0])
 			w.Header().Set("Cache-Control", fmt.Sprintf("max-age=%d", maxAge))
 		}
+		responseHeaderParams, ok := r.URL.Query()["headers"]
+		if ok {
+			for _, header := range responseHeaderParams {
+				h := strings.Split(header, ":")
+				w.Header().Set(h[0], h[1])
```

Comment:
It might be wise to strings.TrimSpace(h[1]). The headers query string is so close to the wire format for headers that I expected it to have a space between the the : and value.  Triming will let both the format used in the e2e test work as well as the canonical wire format.